### PR TITLE
Update add item implementation segment

### LINKED ISSUE #60: Update DG to reflect new Item fields
Update DG to reflect the following field addition to Items: 

- Quantity
- Description

May be relevant to the following sections: 

- [x] addi

### PR DESCRIPTION
**Pull request by Jiacheng-y**
In this PR, I've updated the segment under implementation for add item and its relevant UML diagrams. #203

I have also deleted the irrelevant UML files for undo and redo that we have inherited from AB3

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -156,44 +156,60 @@ This section describes some noteworthy details on how certain features are imple
...
+* quantity (prefixed by `q/`) 
+* description (prefixed by `d/`)
+* sell price (prefixed by `sp/`)
+* cost price (prefied by `cp`)
 
-Step 2. The user executes `addi i/keys q/10 d/Metallic keys sp/15.50 cp/10 t/Utilities` command to add 10 `keys` to item list in TrackO.
-The `addi` command creates an `AddItemCommandParser` which checks the necessary input arguments for item name (prefixed by `i/`), 
-quantity (prefixed by `q/`), description (prefixed by `d/`), selling price (prefixed by `sp/`), cost price (prefixed by `cp/`) 
-are present before parsing the arguments into an `AddItemCommand` object. Tags, prefixed by `t/`, are optional.
+are present before parsing the arguments into an `AddItemCommand` object.
```
Comment (janelleljt) [inline docs/DeveloperGuide.md]: Here it feels a bit awkward that the sentence is being obstructed by the points. Maybe can try something like

```
The `addi` command creates an `AddItemCommandParser` which checks that the following input arguments are present before parsing the arguments into an `AddItemCommand` object:
* item name (prefixed by `i/`)
* quantity (prefixed by `q/`) 
* description (prefixed by `d/`)
* sell price (prefixed by `sp/`)
* cost price (prefied by `cp`)
```

**Review by janelleljt — APPROVED** [review]
Other than some minor changes, looks good!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (Jiacheng-y) [reply docs/DeveloperGuide.md]: ah thanks for the suggestion! I've made the changes

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -156,44 +156,60 @@ This section describes some noteworthy details on how certain features are imple
...
+Tags (which are prefixed by `t/`) are optional inputs and will be parsed into the `AddItemCommand` 
+as well if they are present.
 
 The `AddItemCommand` calls `Model#addItem()` to add the item and its corresponding quantity into the items list.
 
-<div markdown="span" class="alert alert-info">:information_source: **Note:** If a command fails its execution, it will not call `Model#addItem()`, so the incomplete item will not be saved to `ItemsList`.
+![AddItemState1](images/developer-guide/AddItemState1.png)
+
+<div markdown="span" class="alert alert-info">:information_source: **Note:** If a command fails its execution, 
+it will not call `Model#addItem()`, so the incomplete item will not be saved to `ItemsList`.
```
Comment (melissaharijanto) [inline docs/DeveloperGuide.md]: i think this should be `InventoryList` instead, maybe can check for other instances like this?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -156,44 +156,60 @@ This section describes some noteworthy details on how certain features are imple
...
 </div>
 
 The following sequence diagram shows how the add item operation works:
 
-_{insert sequence diagram}_
+![AddItemSequenceDiagram](images/developer-guide/AddItemSequenceDiagram.png)
 
-<div markdown="span" class="alert alert-info">:information_source: **Note:** The lifeline for `AddItemCommand` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+<div markdown="span" class="alert alert-info">:information_source: **Note:** The lifeline for `AddItemCommand` should 
+end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
```
Comment (melissaharijanto) [inline docs/DeveloperGuide.md]: also I think for the rest, destroy marker is typed in italics, so maybe want to do that here as well to be consistent?

**Review by melissaharijanto — CHANGES_REQUESTED** [review]

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (Jiacheng-y) [reply docs/DeveloperGuide.md]: Edited, thank you :)

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (Jiacheng-y) [reply docs/DeveloperGuide.md]: added, thank you :)

**Review by melissaharijanto — APPROVED** [review]

**Pull request merged**